### DISCUSSION TITLE
Input validation for queue URL

### DISCUSSION THREAD

**Discussion by DanielAnfelt** [bbc/sqs-consumer#524] [Feature Request]
### Background

A QueueUrl will always have a rather specific format, meaning that if you send in an arbitrary string, your consumer would emit an error on every polling, rather than crashing the service and giving up.

Given that we know this format, it would be possible to just not allow instantiating the Consumer with an apparently invalid URL, thus possibly preventing a few misconfigurations.

The downside would be that you need to provide a seemingly valid URL when setting up the Consumer for a unit test or something like that. 

### Objectives

1. Preventing accidental misconfiguration of the Consumer, where you end up with a poller that runs but that can never do any work.

**Comment by nicholasgriffintn** [comment]
Thanks, this is one to think on. When using the AWS services directly, the URL should always be more or less the same yes, however, you can use a custom endpoint with SQS, so there's thought on how that could be done with validation.
